src/forms/create-google-files.ts

```diff
@@ -0,0 +1,537 @@
+    if (!AppFormValidator.safeParse(form).success) {
+        throw new Exception(ExceptionType.MARKDOWN, i18nObj.__('general.error-validation-form'), call);
+    }
+
+    return form;
+}
+
+export async function createGoogleDocSubmit(call: ExtendedAppCallRequest): Promise<string> {
+    const i18nObj = configureI18n(call.context);
+    const mattermostUrl: string = call.context.mattermost_site_url!;
+    const userAccessToken: string = call.context.acting_user_access_token!;
+    const actingUserId: string = call.context.acting_user.id!;
+    const botUserId: string = call.context.bot_user_id!;
+    const values = call.values as CreateFileForm;
+
+    const mattermostOpts: MattermostOptions = {
+        mattermostUrl,
+        accessToken: userAccessToken,
+    };
+    const mmClient: MattermostClient = new MattermostClient(mattermostOpts);
+
+    const docs = await getGoogleDocsClient(call);
+    const params = {
+        requestBody: {
+            title: values.google_file_title,
+        },
+    };
+    const newDoc = await tryPromise<Schema$Document>(docs.documents.create(params), ExceptionType.TEXT_ERROR, i18nObj.__('general.google-error'), call);
+    if (!newDoc?.documentId) {
+        throw new Exception(ExceptionType.TEXT_ERROR, i18nObj.__('general.google-error'), call);
+    }
+
+    const drive = await getGoogleDriveClient(call);
+    const paramExport: Params$Resource$Files$Get = {
+        fileId: newDoc.documentId,
+        fields: 'webViewLink,id,owners,permissions,name,iconLink,thumbnailLink,createdTime',
+    };
+
+    const file = await tryPromise<Schema$File>(drive.files.get(paramExport), ExceptionType.TEXT_ERROR, i18nObj.__('general.google-error'), call);
+    const owner: Schema$User | undefined = head(file.owners);
+
+    let channelId: string = call.context.channel?.id;
+    if (!values.google_file_will_share) {
+        const channel: Channel = await mmClient.createDirectChannel([botUserId, actingUserId]);
+        channelId = channel.id;
+    }
+
+    const date = moment(file?.createdTime).format('MMM Do, YYYY');
+
+    const post: PostCreate = {
+        message: values.google_file_message || '',
+        channel_id: channelId,
+        props: {
+            attachments: [
+                {
+                    author_name: `${owner?.displayName}`,
+                    author_icon: `${owner?.photoLink}`,
+                    title: `${file.name}`,
+                    title_link: `${file.webViewLink}`,
+                    footer: i18nObj.__('create-binding.response.footer', { date }),
+                    footer_icon: `${file.iconLink}`,
+                    fields: [],
+                    actions: [],
+                },
+            ],
+        },
+    };
+    await mmClient.createPost(post);
+
+    const shareFile: ShareFileFunction = SHARE_FILE_ACTIONS[values.google_file_access.value];
+    if (shareFile) {
+        await shareFile(call, file, channelId);
+    }
```

Comment:
Currently the logic to check `configClient?.ShowEmailAddress;` is only present while displaying the Form to users. However a user can easily modify the value of `google_file_access` parameter sent to the path `/create/document/submit` API endpoint and still be able to retrieve the email addresses. The same logic to check `configClient?.ShowEmailAddress;` should also be applied here.